File: src/main/java/seremis/geninfusion/inventory/slot/FilteredSlot.java
package seremis.geninfusion.inventory.slot;

import net.minecraft.inventory.IInventory;
import net.minecraft.item.ItemStack;

public class FilteredSlot extends GISlot {

    public ItemStack filter;
    public int slotStackLimit;

    public FilteredSlot(IInventory inventory, int id, int x, int y, ItemStack filter, int slotStackLimit) {
        super(inventory, id, x, y);
        this.filter = filter;
        this.slotStackLimit = slotStackLimit;
    }

    @Override
    public boolean isItemValid(ItemStack stack) {
        if(filter != null) {
            if(stack.isItemEqual(filter)) {
                return super.isItemValid(stack);
            }
        }
        return false;
    }

    @Override
    public int getSlotStackLimit() {
        return slotStackLimit;
    }
}

File: src/main/java/seremis/geninfusion/inventory/slot/FilteredSlot.scala
package seremis.geninfusion.inventory.slot

import net.minecraft.inventory.IInventory
import net.minecraft.item.ItemStack

class FilteredSlot(inventory: IInventory, id: Int, x: Int, y: Int, filter: ItemStack, slotStackLimit: Int) extends GISlot(inventory, id, x, y) {

    override def isItemValid(stack: ItemStack): Boolean = {
        if (filter != null) {
            if (stack.isItemEqual(filter)) {
                return super.isItemValid(stack)
            }
        }
        false
    }

    override def getSlotStackLimit: Int = slotStackLimit
}